judo-ui-react/src/main/resources/actor/src/layout/BottomMenu/index.tsx.hbs

```diff
@@ -0,0 +1,357 @@
+    setBottomMenuItems(filteredMenus);
+  }, [{{# if application.authentication }}principal{{/ if }}]);
+
+  let lastItemIndex = useMemo(() => bottomMenuItems.length - 1, [bottomMenuItems]);
+
+  const handlePerform = (event: React.MouseEvent<HTMLElement>, item: NavItemType) => {
+    console.log(item.title);
+    event.preventDefault();
+    event.stopPropagation();
+    if (typeof item.onClick === 'function') {
+      try {
+        item.onClick();
+      } catch (e) {
+        console.error(e);
+      }
+    } else if (item.url!.startsWith('http')) {
+      externalNavigate(item.url!);
+    } else {
+      clearNavigate(item.url!);
+    }
+  };
+
+  const popupMenuItemAction = (item: NavItemType) => {
+    switch (item.type) {
+      case 'group':
+        return (
+          <NavCollapse
+            key={item.id}
+            menu={item}
+            level={2}
+            render={true}
+            placement="auto"
+            arrow={false}
+            mouseOver={false}
+          />
+        );
+      case 'collapse':
+        return (
+          <NavCollapse
+            key={item.id}
+            menu={item}
+            level={2}
+            render={true}
+            placement="auto"
+            arrow={false}
+            mouseOver={false}
+          />
+        );
+      case 'item':
+        return <NavItem key={item.id} item={item} level={2} />;
+      default:
+        return (
+          <Typography key={item.id} variant="h6" color="error" align="center">
+            Menu Items Error
+          </Typography>
+        );
+    }
+  };
+
+  const handlePopupOpen = (event: React.MouseEvent<HTMLElement>, item: NavItemType) => {
+    setMouseX(event.clientX);
+    setMouseY(event.clientY);
+    setSelectedBottomMenuItem(item);
+    setSelectedBottomOther(false);
+    setSubMenuItems(
+      item.children?.map((menu) => {
+        return popupMenuItemAction(menu);
+      }),
+    );
+    setOpen(true);
+  };
+
+  const handleOtherPopupOpen = (event: React.MouseEvent<HTMLElement>) => {
+    setMouseX(event.clientX);
+    setMouseY(event.clientY);
+    setSelectedBottomMenuItem(undefined);
+    setSelectedBottomOther(true);
+    setSubMenuItems([
+      ...bottomMenuItems.slice(3, lastItemIndex + 1).map((item) => {
+        return popupMenuItemAction(item);
+      }),
+    ]);
+    setOpen(true);
+  };
+
+  const handleClose = () => {
+    setOpen(false);
+    setSubMenuItems(undefined);
+    setSelectedBottomMenuItem(undefined);
+    setSelectedBottomOther(false);
+  };
+
+  const [mouseX, setMouseX] = useState(0);
+  const [mouseY, setMouseY] = useState(0);
+
+  const itemAction = (item: NavItemType) => {
+    switch (item.type) {
+      case 'group':
+        return (
+          <BottomNavigationAction
+            aria-describedby="submenu"
+            showLabel={true}
+            key={item.id + 'NavAction'}
```

Comment:
If this `key` is for React, then we don't need it, because the component is not iterated inside a .map().